src/components/Timeline.vue

```diff
@@ -23,7 +23,20 @@ import { computed, PropType, defineComponent } from 'vue';
 import 'vue-slider-component/theme/default.css'
 import { TIMELINE_YEARS,BEGIN_MODELED_YEAR } from '@/models/constants';
 import { useCatastropheStore } from '@/stores/catastrophes';
+import { useStatisticStore } from '@/stores/statistics';
 import { CatastropheFilter } from '@/models/catastrophes';
+import { temperatureGradient, getGradientColourIndex,colourToHex } from '@/utils/colours';
+
+ function generateTemperatureGradient(statisticStore: any, district: number, progressPercent: number): String[] {
```

Comment:
`statisticStore` a un type, faque tu devrais pas mettre `any`. Ce type est très obtus, par contre, tu peux l'obtenir avec `ReturnType` et `typeof`:

    statisticStore: ReturnType<typeof useStatisticStore>

Ça va te révéler que tu gères pas le fait que `temp_delta` peut être undefined dans certains cas.

Cela étant dit, cette fonction n'est utilisée qu'à un seul endroit, donc tu pourrais juste l'inliner et utiliser `this.statisticStore` directement.

Reply:
Ah je cherchais justement le type du store, sans succès. J'y ai été pour ~yolo~ any. Merci du truc

Reply:
Dans quel cas le temp_delta est undefined? Je pensais que 1990 était le seul cas mais ça a été corrigé dans cette pr

Reply:
Dans les cas ou une région a pas de température pour une année en particulier. Tu remarqueras que techniquement, toutes les stats sur une région sont optionnelles. Je préfère ça que de rentrer 0 dans une stat, parce que comme ça on l'affiche pas dans le UI.